[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[32mINFO    [0m pytest_operator.plugin:plugin.py:753 Adding model github-pr-3074c-microk8s:test-upgrade-first-kmv4 on cloud microk8s
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHSession received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHTunneler received from the controller
[32mINFO    [0m test_upgrade_first:test_upgrade_first.py:30 Deploying vault from Charmhub
[32mINFO    [0m juju.model:__init__.py:2301 Deploying ch:amd64/noble/vault-k8s-353
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s (waiting for exactly 3 units, current : 1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [executing] waiting: Waiting to be able to connect to vault unit
  vault-k8s/1 [allocating] waiting: installing agent
  vault-k8s/2 [allocating] waiting: installing agent
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/1 [idle] waiting: Waiting for CA certificate to be accessible in the charm
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/2 [idle] waiting: Waiting for CA certificate to be accessible in the charm
[32mINFO    [0m helpers:helpers.py:215 Vault initialized
[32mINFO    [0m tests.integration.vault:vault.py:94 Unsealed vault unit: https://10.1.63.208:8200.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m tests.integration.vault:vault.py:94 Unsealed vault unit: https://10.1.63.216:8200.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m tests.integration.vault:vault.py:94 Unsealed vault unit: https://10.1.63.215:8200.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [executing] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m helpers:helpers.py:246 Charm authorized
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] blocked: Please authorize charm (see `authorize-charm` action)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] blocked: Please authorize charm (see `authorize-charm` action)
  vault-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] active: 
  vault-k8s/2 [idle] active: 
[32mINFO    [0m test_upgrade_first:test_upgrade_first.py:53 Refreshing vault from built charm
[32mINFO    [0m test_upgrade_first:test_upgrade_first.py:56 Waiting for vault to be blocked after refresh
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] active: 
  vault-k8s/2 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [executing] maintenance: 
  vault-k8s/2 [executing] waiting: Waiting for vault to be available
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] maintenance: 
  vault-k8s/1 [idle] waiting: Waiting for vault to be available
  vault-k8s/2 [idle] waiting: Waiting for vault to be available
[32mINFO    [0m tests.integration.vault:vault.py:94 Unsealed vault unit: https://10.1.63.219:8200.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m tests.integration.vault:vault.py:94 Unsealed vault unit: https://10.1.63.218:8200.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m tests.integration.vault:vault.py:94 Unsealed vault unit: https://10.1.63.217:8200.
[32mINFO    [0m tests.integration.vault:vault.py:76 Vault unit is unsealed.
[32mINFO    [0m test_upgrade_first:test_upgrade_first.py:69 Waiting for vault to be active after refresh
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  vault-k8s/0 [idle] active: 
  vault-k8s/1 [idle] active: 
  vault-k8s/2 [idle] active:
[32mINFO    [0m pytest_operator.plugin:plugin.py:937 Model status:

Model                    Controller                Cloud/Region        Version  SLA          Timestamp
test-upgrade-first-kmv4  github-pr-3074c-microk8s  microk8s/localhost  3.6.8    unsupported  14:06:56Z

App        Version  Status  Scale  Charm      Channel  Rev  Address        Exposed  Message
vault-k8s           active      3  vault-k8s             0  10.152.183.88  no       

Unit          Workload  Agent  Address      Ports  Message
vault-k8s/0*  active    idle   10.1.63.219         
vault-k8s/1   active    idle   10.1.63.218         
vault-k8s/2   active    idle   10.1.63.217         

[32mINFO    [0m pytest_operator.plugin:plugin.py:943 Juju error logs:

unit-vault-k8s-2: 14:05:13 ERROR unit.vault-k8s/2.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-2.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fb36e952210>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-2: 14:05:14 ERROR unit.vault-k8s/2.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-2.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f5ab65f2c30>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-2: 14:05:16 ERROR unit.vault-k8s/2.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-2.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f4f791306e0>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-2: 14:05:17 ERROR unit.vault-k8s/2.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-2.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fada71f46b0>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-2: 14:05:19 ERROR unit.vault-k8s/2.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-2.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f0d018daa80>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-1: 14:05:33 ERROR unit.vault-k8s/1.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-1.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fa21ae2a2d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-1: 14:05:34 ERROR unit.vault-k8s/1.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-1.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fde99cfcef0>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-1: 14:05:36 ERROR unit.vault-k8s/1.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-1.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fde135f0530>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-1: 14:05:37 ERROR unit.vault-k8s/1.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-1.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f1dc6ae4740>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-1: 14:05:39 ERROR unit.vault-k8s/1.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-1.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7effb092a900>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-vault-k8s-0: 14:05:53 ERROR unit.vault-k8s/0.juju-log [vault_client] Error while checking Vault health status: HTTPSConnectionPool(host='vault-k8s-0.vault-k8s-endpoints.test-upgrade-first-kmv4.svc.cluster.local', port=8200): Max retries exceeded with url: /v1/sys/health (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f77d94099a0>: Failed to establish a new connection: [Errno 111] Connection refused'))

[32mINFO    [0m pytest_operator.plugin:plugin.py:1049 Resetting model test-upgrade-first-kmv4...
[32mINFO    [0m pytest_operator.plugin:plugin.py:1038    Destroying applications vault-k8s
[32mINFO    [0m pytest_operator.plugin:plugin.py:1054 Not waiting on reset to complete.
[32mINFO    [0m pytest_operator.plugin:plugin.py:1025 Forgetting model main...